lib/JSON/PP.pm

```diff
@@ -1493,7 +1498,20 @@ BEGIN {
 $JSON::PP::true  = do { bless \(my $dummy = 1), "JSON::PP::Boolean" };
 $JSON::PP::false = do { bless \(my $dummy = 0), "JSON::PP::Boolean" };
 
-sub is_bool { blessed $_[0] and ( $_[0]->isa("JSON::PP::Boolean") or $_[0]->isa("Types::Serialiser::BooleanBase") or $_[0]->isa("JSON::XS::Boolean") ); }
+sub is_bool {
+  if (blessed $_[0]) {
+    return (
+      $_[0]->isa("JSON::PP::Boolean")
+      or $_[0]->isa("Types::Serialiser::BooleanBase")
+      or $_[0]->isa("JSON::XS::Boolean")
+    );
+  }
+  elsif (CORE_BOOL) {
+    BEGIN { CORE_BOOL and warnings->unimport('experimental::builtin') }
+    return builtin::is_bool($_[0]);
+  }
+  return !!0;
+}
 
 sub true  { $JSON::PP::true  }
```

Comment:
It there a reason for not making these two functions return native bools?

Reply:
See #63. It would be unwise to use native bools unless the application is equipped to deal with them everywhere.

Reply:
`!!0` is a native bool, and both `is_bool` and `isa` should be returning native bools.

Reply:
Or, if you were referring to the true and false functions, that is because many things expect the bools coming from JSON::PP to be JSON::PP::Boolean objects. For now, I've left those unchanged for backwards compatibility. Also, this PR is only meant to deal with encoding. #63 exists to start dealing with decoding.